crypto/s2n_rsa.c

```diff
@@ -93,18 +93,31 @@ int s2n_rsa_private_key_free(struct s2n_rsa_private_key *key)
     return 0;
 }
 
+static BIGNUM *s2n_rsa_get_modulus(RSA *rsa)
```

Comment:
How about instead of a modulus, we have a "s2n_rsa_check_modulus" ? That way we can port to non-OpenSSL libraries more easily. BIGNUM is very OpenSSL specific. But the act of checking a modulus is general.